scripts/js-analyze.js

```diff
@@ -2103,6 +2150,146 @@ function analyzeHTML(filename)
   parser2.processEventListeners();
 }
 
+class CSSAnalyzer {
```

Comment:
We already have the rust CSS parser which we use to tokenize the CSS. Maybe we should use it to analyze the CSS as well? Should be trivial-ish.

But not objecting to using js-analyze I guess, it's just probably less efficient.

Reply:
I think the idea of doing this from within js-analyze is that we also [use it for the HTML analysis too](https://github.com/mozsearch/mozsearch/blob/0f89a38f08c8cdc4375291feaf6ee11c5736961a/scripts/html-analyze.sh#L17) and there is synergy for HTML/JS/CSS analysis to all be using the same codebase because of how HTML can include CSS and JS inline and how an unfortunate reality is that JS may also potentially contain inline HTML and CSS as well.

In terms of using tree-sitter-css, that was my suggestion in a DM conversation where that topic didn't really need to be in DM, so I'll include my message below.  The way we end up using the parser here doesn't really result in any direct synergies that I was thinking of, though, and potentially using tree-sitter for CSS tokenization (for consistency with blame history and the potential for cheap position:sticky) is still a ways off, so I think the change to cssparser makes sense (and provides more synergy for the current tokenizer situation).

The info dump messages:

> for tree-sitter-css, the main reason I find it interesting is, and I think I mentioned something related this recently... the [tree-sitter query syntax (docs)](https://tree-sitter.github.io/tree-sitter/using-parsers#query-syntax) is quite powerful and enables the ability to treat different languages somewhat generically.  So for example for the hyperblame effort, the [C++ query support](https://github.com/mozsearch/mozsearch/blob/master/tools/languages/tokenizer_queries/cpp.scm) helps us map from the AST to providing nesting semantics for C++, but then we also are able to handle [python](https://github.com/mozsearch/mozsearch/blob/master/tools/languages/tokenizer_queries/python.scm), [rust](https://github.com/mozsearch/mozsearch/blob/master/tools/languages/tokenizer_queries/rust.scm), and [TS/JS](https://github.com/mozsearch/mozsearch/blob/master/tools/languages/tokenizer_queries/typescript.scm) without much per-language glue code.
>
> There is an interesting potential to use (different) queries and some additional configuration to let a more generic tool built on tree-sitter potentially provide the best-effort indexing in various cases.  Like how [the tree-sitter-cpp has a tags.scm](https://github.com/tree-sitter/tree-sitter-cpp/blob/master/queries/tags.scm) for tags indexing, and that same mechanism can then [be used for ocaml](https://github.com/tree-sitter/tree-sitter-ocaml/blob/master/queries/tags.scm) as one example of many. https://tree-sitter.github.io/tree-sitter/code-navigation-systems documents the tags used for the tags command.
>
> There is no tags.scm for tree-sitter-css right now, though, just [highlights](https://github.com/tree-sitter/tree-sitter-css/blob/master/queries/highlights.scm) for the [tree-sitter syntax highlighting support](https://tree-sitter.github.io/tree-sitter/syntax-highlighting).  I don't know how well / if the tags semantics would translate well for searchfox or not.  (In general it's nice when we can have precise indexing as opposed to the syntactically based best-effort like tree-sitter would provide.)

> But CSS is also such a fundamental part of the web with semantics very unlike normal program languages, so it could also make sense to not try and use the more generic tree-sitter-css at all for semantic indexing.  Especially since it does seem likely a lot of (existing) CSS tooling will need token position information anyways for responsible error reporting.
>
> (And we can still use tree-sitter-css for the hyperblame implementation.  Like we're using tree-sitter-mozcpp for hyperblame even though we don't use that for our actual indexer/analysis.)